In [83]:
import pandas as pd
import numpy as np

In [84]:
auto_mpg_df = pd.read_csv('autompg.csv')
# split the data frame based on country of origin
auto_mpg_us_df = auto_mpg_df[auto_mpg_df['origin'] == 1]
auto_mpg_japan_df = auto_mpg_df[auto_mpg_df['origin'] == 3]
auto_mpg_germany_df = auto_mpg_df[auto_mpg_df['origin'] == 2]

In [85]:
# split based on number of cylinders
# for US cars
auto_mpg_us_4_cylinders_df = auto_mpg_us_df[auto_mpg_us_df['cylinders'] == 4]
auto_mpg_us_6_cylinders_df = auto_mpg_us_df[auto_mpg_us_df['cylinders'] == 6]
# for Japanese cars
auto_mpg_japan_4_cylinders_df = auto_mpg_japan_df[auto_mpg_japan_df['cylinders'] == 4]
auto_mpg_japan_6_cylinders_df = auto_mpg_japan_df[auto_mpg_japan_df['cylinders'] == 6]
# for German cars
auto_mpg_germany_4_cylinders_df = auto_mpg_germany_df[auto_mpg_germany_df['cylinders'] == 4]
auto_mpg_germany_6_cylinders_df = auto_mpg_germany_df[auto_mpg_germany_df['cylinders'] == 6]

We know that Japanese cars are more fuel efficient than US and German cars now. However this data is for cars between the years of 1970 and 1982.
Let's examine if the data shows any statistical significance.

##Test Case I
###For 4 cylinder cars 
####Null hypothesis:
Mean_mpg_Japanese - Mean_mpg_German > 0

and 

Mean_mpg_German - Mean_mpg_US > 0
####Alternative hypothesis
Mean_mpg_Japanese - Mean_mpg_German <= 0

and

Mean_mpg_German - Mean_mpg_US <= 0

In [86]:
mean_mpg_Japanese_4_cyl = auto_mpg_japan_4_cylinders_df['mpg'].mean()
std_mpg_Japanese_4_cyl = auto_mpg_japan_4_cylinders_df['mpg'].std()
print mean_mpg_Japanese_4_cyl, std_mpg_Japanese_4_cyl

31.5956521739 5.43578650575


In [87]:
mean_mpg_German_4_cyl = auto_mpg_germany_4_cylinders_df['mpg'].mean()
std_mpg_German_4_cyl = auto_mpg_germany_4_cylinders_df['mpg'].std()
print mean_mpg_German_4_cyl, std_mpg_German_4_cyl

28.106557377 6.29107486451


In [88]:
mean_mpg_US_4_cyl = auto_mpg_us_4_cylinders_df['mpg'].mean()
std_mpg_US_4_cyl = auto_mpg_us_4_cylinders_df['mpg'].std()
print mean_mpg_US_4_cyl, std_mpg_US_4_cyl

28.0130434783 4.56659588249


Parameter of interest = mean1 - mean2

So SE = sqrt(SE1 x SE1/N1 + SE2 x SE2/N2)

In [89]:
se_japan_german = np.sqrt(std_mpg_Japanese_4_cyl**2/len(auto_mpg_japan_4_cylinders_df) + std_mpg_German_4_cyl**2/len(auto_mpg_germany_4_cylinders_df))
se_japan_german

1.0378063948823995

So for a 95% confidence interval of the mean

In [90]:
diff_japan_german_upper = (mean_mpg_Japanese_4_cyl - mean_mpg_German_4_cyl) + 1.96 * se_japan_german
diff_japan_german_lower = (mean_mpg_Japanese_4_cyl - mean_mpg_German_4_cyl) - 1.96 * se_japan_german
print "Confidence interval is between", diff_japan_german_upper, diff_japan_german_lower

Confidence interval is between 5.52319533083 1.45499426289


This means we are 95% confident that the difference in mean mpg between Japanese and German cars lies between 5.52 and 1.45

Lets also calculate the p value

In [91]:
z_japan_germany = ((mean_mpg_Japanese_4_cyl - mean_mpg_German_4_cyl) - 0)/se_japan_german
z_japan_germany

3.3619900725888554

We are interested in the right side of the normal curve which is found from the tables to be 0.9996

###This means we are 99.96% confident that our Null hypothesis is correct for Japanese and German cars

Now lets compare 4 cylinder German cars and US cars

In [92]:
se_german_us = np.sqrt(std_mpg_German_4_cyl**2/len(auto_mpg_germany_4_cylinders_df) + std_mpg_US_4_cyl**2/len(auto_mpg_us_4_cylinders_df))
se_german_us

0.97521405181200982

So for a 95% confidence interval of the mean

In [93]:
diff_us_german_upper = (mean_mpg_German_4_cyl - mean_mpg_US_4_cyl) + 1.96 * se_german_us
diff_us_german_lower = (mean_mpg_German_4_cyl - mean_mpg_US_4_cyl) - 1.96 * se_german_us
print "Confidence interval is between", diff_us_german_upper, diff_us_german_lower

Confidence interval is between 2.00493344034 -1.81790564276


This is by no means a conclusive answer and we definitely need to look at the p value

In [94]:
z_us_germany = ((mean_mpg_German_4_cyl - mean_mpg_US_4_cyl) - 0)/se_german_us
z_us_germany

0.095890639203308647

We are interested in the right side of the normal curve which is found from the tables to be 0.5359

###This means we are 53.59% confident that our Null hypothesis is correct for US and German cars



##Test Case II
###For 6 cylinder cars 
####Null hypothesis:
Mean_mpg_Japanese - Mean_mpg_German > 0

and 

Mean_mpg_German - Mean_mpg_US > 0
####Alternative hypothesis
Mean_mpg_Japanese - Mean_mpg_German <= 0

and

Mean_mpg_German - Mean_mpg_US <= 0

In [95]:
mean_mpg_Japanese_6_cyl = auto_mpg_japan_6_cylinders_df['mpg'].mean()
std_mpg_Japanese_6_cyl = auto_mpg_japan_6_cylinders_df['mpg'].std()
print mean_mpg_Japanese_6_cyl, std_mpg_Japanese_6_cyl

23.8833333333 4.95193564848


In [96]:
mean_mpg_German_6_cyl = auto_mpg_germany_6_cylinders_df['mpg'].mean()
std_mpg_German_6_cyl = auto_mpg_germany_6_cylinders_df['mpg'].std()
print mean_mpg_German_6_cyl, std_mpg_German_6_cyl

20.1 7.07436687391


Parameter of interest = mean1 - mean2

So SE = sqrt(SE1 x SE1/N1 + SE2 x SE2/N2)

In [97]:
se_japan_german = np.sqrt(std_mpg_Japanese_6_cyl**2/len(auto_mpg_japan_6_cylinders_df) + std_mpg_German_6_cyl**2/len(auto_mpg_germany_6_cylinders_df))
se_japan_german

4.0741393092420282

So for a 95% confidence interval of the mean

In [98]:
diff_japan_german_upper = (mean_mpg_Japanese_6_cyl - mean_mpg_German_6_cyl) + 1.96 * se_japan_german
diff_japan_german_lower = (mean_mpg_Japanese_6_cyl - mean_mpg_German_6_cyl) - 1.96 * se_japan_german
print "Confidence interval is between", diff_japan_german_upper, diff_japan_german_lower

Confidence interval is between 11.7686463794 -4.20197971278


This is by no means a conclusive answer and we definitely need to look at the p value

In [99]:
z_japan_germany = ((mean_mpg_Japanese_6_cyl - mean_mpg_German_6_cyl) - 0)/se_japan_german
z_japan_germany

0.92862149429966268

From tables p value os 0.8212

###So the Null hypothesis for Japanese and German still holds for 6 cylinders

Now lets compare 6 cylinder engines of German and US make

In [100]:
mean_mpg_US_6_cyl = auto_mpg_us_6_cylinders_df['mpg'].mean()
std_mpg_US_6_cyl = auto_mpg_us_6_cylinders_df['mpg'].std()
print mean_mpg_US_6_cyl, std_mpg_US_6_cyl

19.6452054795 3.39464615632


In [101]:
se_german_us = np.sqrt(std_mpg_German_6_cyl**2/len(auto_mpg_germany_6_cylinders_df) + std_mpg_US_6_cyl**2/len(auto_mpg_us_6_cylinders_df))
se_german_us

3.559427553623939

So for a 95% confidence interval of the mean

In [102]:
diff_us_german_upper = (mean_mpg_German_6_cyl - mean_mpg_US_6_cyl) + 1.96 * se_german_us
diff_us_german_lower = (mean_mpg_German_6_cyl - mean_mpg_US_6_cyl) - 1.96 * se_german_us
print "Confidence interval is between", diff_us_german_upper, diff_us_german_lower

Confidence interval is between 7.43127252565 -6.52168348455


This is by no means a conclusive answer and we definitely need to look at the p value

In [103]:
z_us_germany = ((mean_mpg_German_6_cyl - mean_mpg_US_6_cyl) - 0)/se_german_us
z_us_germany

0.12777181546648197

The right hand side of the curve yields a p value of 0.5478

###So again the Null hypothesis holds for 6 cylinder engines of German and US make

In [105]:
auto_mpg_us_4_cylinders_df

mpg  cylinders  displacement  horsepower  weight  acceleration  \
30   28.0          4         140.0          90    2264          15.5   
45   22.0          4         140.0          72    2408          19.0   
48   23.0          4         122.0          86    2220          14.0   
55   26.0          4          91.0          70    1955          20.5   
57   25.0          4          97.5          80    2126          17.0   
59   20.0          4         140.0          90    2408          19.5   
60   21.0          4         122.0          86    2226          16.5   
79   22.0          4         122.0          86    2395          16.0   
82   28.0          4          98.0          80    2164          15.0   
108  21.0          4         140.0          72    2401          19.5   
111  19.0          4         122.0          85    2310          18.5   
128  26.0          4         122.0          80    2451          16.5   
130  25.0          4         140.0          75    2542          17.0   
144  28.0          4          90.0          75    2125          14.5   
166  23.0          4         140.0          83    2639          17.0   
168  23.0          4         140.0          78    2592          18.5   
182  25.0          4         140.0          92    2572          14.9   
183  26.0          4          98.0          79    2255          17.7   
193  29.0          4          85.0          52    2035          22.2   
194  24.5          4          98.0          60    2164          22.1   
204  26.5          4         140.0          72    2565          13.6   
215  30.0          4         111.0          80    2155          14.8   
217  25.5          4         122.0          96    2300          15.5   
232  24.5          4         151.0          88    2740          16.0   
234  25.5          4         140.0          89    2755          15.8   
235  30.5          4          98.0          63    2051          17.0   
236  33.5          4          98.0          83    2075          15.9   
243  36.1          4          98.0          66    1800          14.4   
253  25.1          4         140.0          88    2720          15.4   
264  30.0          4          98.0          68    2155          16.5   
..    ...        ...           ...         ...     ...           ...   
306  33.5          4         151.0          90    2556          13.2   
309  32.1          4          98.0          70    2120          15.5   
311  28.0          4         151.0          90    2678          16.5   
312  26.4          4         140.0          88    2870          18.1   
313  24.3          4         151.0          90    3003          20.1   
321  27.9          4         156.0         105    2800          14.4   
334  27.2          4         135.0          84    2490          15.7   
335  26.6          4         151.0          84    2635          16.4   
336  25.8          4         156.0          92    2620          14.4   
338  30.0          4         135.0          84    2385          12.9   
340  39.0          4          86.0          64    1875          16.4   
346  34.7          4         105.0          63    2215          14.9   
347  34.4          4          98.0          65    2045          16.2   
348  29.9          4          98.0          65    2380          20.7   
362  28.0          4         112.0          88    2605          19.6   
363  27.0          4         112.0          88    2640          18.6   
364  34.0          4         112.0          88    2395          18.0   
365  31.0          4         112.0          85    2575          16.2   
366  29.0          4         135.0          84    2525          16.0   
367  27.0          4         151.0          90    2735          18.0   
368  24.0          4         140.0          92    2865          16.4   
372  38.0          4         105.0          63    2125          14.7   
373  36.0          4          98.0          70    2125          17.3   
382  26.0          4         156.0          92